Populate rep

nadjano opened the pull request:
Populated rep with scripts, envs and Snakefile to create reference library at $DECONV_REF. For running pipeline,  run_deconvolution_reference_creation.sh with configuration variables and config.yaml with scxa accessions is required.
It should be possible to create these references for any scxa accession with cell type ontology labels and organism part ontology labels, but we have to manually decide which accession make sense to include. 

Main steps of the pipeline are:
- copy input files for ANND and non ANND experiments (this is still a bit problematic for ANND accession as they are not completely curated)
- create SeuratObjects per scxa experiment
- split SeuratObjects into organism.parts (if multiple)
- reduce number of cell types (if too many)
- create three kind of references per SeuratObject which are required by the deconvolution tools
- scale one of these references (input for DWLS)
- plot UMAPs to visualise references and check cell type label reduction
- make a summary file of which references where created from which accession

The output of this pipeline are therefore:
- deconvolution reference lib with three files per scxa accession + organism part
- one summary file
- one png file  per scxa accession + organism part with two UMAP plots for quality control

It can happen that more than one of the scxa experiments has a specific organism part, therefore we must decide from which scxa experiment the reference looks better (in terms of number of cells, cell type labels ...). To exclude organism parts from accessions they can be included in the config.yaml to avoid them being created.

nadjano committed:
add content

nadjano committed:
added scripts

nadjano committed:
addedn ontology lists

nadjano committed:
added config file

nadjano committed:
added envs

nadjano committed:
removed config file

nadjano committed:
added date

nadjano committed:
reduce mem

nadjano committed:
change location of ontology file

nadjano committed:
Added comments for functions

nadjano committed:
Removed commented code

nadjano committed:
added comment

nadjano committed:
Added some comments and removed whitespace

nadjano committed:
added comment

nadjano committed:
remove whitespace

nadjano committed:
Added comments

nadjano committed:
remove whitespace

nadjano committed:
added content

nadjano committed:
Update README.md

nadjano committed:
Update README.md

nadjano committed:
Update Snakefile

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+rule all:
+    input:
+        #get_tissues_per_accession()
+        config['deconv_ref'] + '/homo_sapiens_summary.tsv'
```
the species here is hardcoded. Could we specify and get also from the config file? In case we want to run for other organisms , such as mus_musculus

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+        #get_tissues_per_accession()
+        config['deconv_ref'] + '/homo_sapiens_summary.tsv'
+        
+#Rule for copying all the project anndatafiles we want to build references from
```
```suggestion
# Rule for copying all the project anndata files we want to build references from
```

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+    """
+    To adjust resources in the rules 
+    attemps = reiterations + 1
+    Max number attemps = 8
```
it doesn't seem to be 8

nadjano committed:
changed name of config file

 to match file on gitlab

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+    else:
+        return mem_avail[attempt-1] * 1000
+
+#Rule all
```
```suggestion
# Rule all
```

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+        if [[ {wildcards.experiment} == *"ANND"* ]]; then
+            cp {input} scxa_input/{wildcards.experiment}
+        else
+            cp {input} scxa_input/{wildcards.experiment}
```
the `cp` command is the same. Why do you need the if/else?

nadjano replied:
Thats ture!

pmb59 commented on README.md:
````diff
@@ -1,2 +1,24 @@
+
+```
+run_deconvolution_reference_creation.sh
+```
````
Probably you could comment about the outputs.

pmb59 replied:
If you are using any code from https://github.com/Functional-Genomics/CATD_snakemake , it would be good add a reference in the end of the README

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+        "envs/createSeuratObject.yaml"
+    resources: mem_mb=get_mem_mb
+    shell:
+        """
```
Please consider adding this to the rules
```suggestion
        """
        set -e # snakemake on the cluster doesn't stop on error when --keep-going is set
        exec &> "{log}"
```

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+        fi
+        """
+
+#Rule for creating SeuratObjects for anndata and non anndata experiments
```
```suggestion
# Rule for creating SeuratObjects for anndata and non anndata experiments
```

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+        """
+
+#Rule to split sc reference into single organism parts
+rule splitByTissue:
```
Please consider adding info like this instead of outer comment
```suggestion
rule splitByTissue:
    """
    Rule to split sc reference into single organism parts.
    """
```

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+    output:
+        temp(config['deconv_ref'] + "/{tissue}_{experiment}_seurat_curated.rds")  
+    conda:
+        "envs/scONTO.yaml"
```
please consider having the conda env definition first in each rule

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+        """
+
+#Rule to generate different kind of references from the SeuratObject
+rule generateReferences:
```
Please explain a bit in the commented part what are the different outputs (C0, C1, phenData)
```suggestion
rule generateReferences:
    """
    Rule to generate different kind of references from the SeuratObject
    """
```

nadjano committed:
Create example.config

added some example to show how config file should be used

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+        """
+
+#Rule to scale sc references
+rule scale_C:
```
could you please rename the rules in a way that the name is more informative (instead of `scale_C`)?
```suggestion
rule scale_C:
    """
    Rule to scale sc references. Scaling process consist of ... 
    """
```

nadjano committed:
Update README.md

nadjano committed:
Update README.md

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+    input:
+        expand(get_tissues_per_accession())
+    output:
+        config['deconv_ref'] + '/homo_sapiens_summary.tsv'
```
the species here is hardcoded. Could we specify and get also from the config file? In case we want to run for other organisms , such as mus_musculus

pmb59 commented on envs/TissueSplit.yaml:
```diff
@@ -0,0 +1,10 @@
+  - bioconda
+
+dependencies:
+  - r-seurat
```
maybe you can pin Seurat to be a version `>=` than the one you are currently using?

pmb59 commented on envs/UMAPplot.yaml:
```diff
@@ -0,0 +1,9 @@
+name: UMAP_plot
+
```
It is fine to get rid of these spaces
```suggestion
```

pmb59 commented on envs/env.yaml:
```diff
@@ -0,0 +1,319 @@
+name: decon_convert
```
the name of this file is `env.yaml`, probably should be changed

pmb59 commented on scripts/makeSummary.R:
```diff
@@ -0,0 +1,46 @@
+}
+
+#read in UBERON ontologys to translate UBERON ids to organism part labels
+obo_file = 'deconvolution-reference-library/files/basic.obo' #download here: http://purl.obolibrary.org/obo/uberon/basic.obo
```
please add to the README how you generated the 2 `.obo` files

pmb59 commented on scripts/convert.R:
```diff
@@ -0,0 +1,45 @@
+## script to read in mtx files from ANND experiments
+## and create SeuratObject
+##
```
```suggestion
```

pmb59 commented on scripts/convert.R:
```diff
@@ -0,0 +1,45 @@
+## and create SeuratObject
+##
+suppressMessages(library(devtools))
+devtools::install_github('cellgeni/sceasy')
```
I assume this can not be added to the conda env.

In that case, you don't want to install every time in the environment, but check first. Something like:

```
if (!require("pacman")) install.packages("pacman")
```

pmb59 commented on scripts/convert.R:
```diff
@@ -0,0 +1,45 @@
+                        sep = '\t', row.names = 1) #row.names should be cellIDs
+
+#create SeuratObject
+C = CreateSeuratObject(counts = counts, meta.data = metadata)
```
please consider renaming following valriables
```
filename_O 
C
t
```

pmb59 commented on scripts/convert.R:
```diff
@@ -0,0 +1,45 @@
+suppressMessages(library(plyr))
+suppressMessages(library(stringr))
+
+args <- commandArgs(trailingOnly = TRUE)
```
You can add a check here:
```suggestion
args <- commandArgs(trailingOnly = TRUE)

if(length(args) != 1) {
   stop("Not correct number of arguments. Please supply 1 argument")
}
```

pmb59 commented on scripts/convert.R:
```diff
@@ -0,0 +1,45 @@
+metadata = read.csv(paste0('scxa_input/',sample,'/', sample, '.cell_metadata.tsv'), 
+                        sep = '\t', row.names = 1) #row.names should be cellIDs
+
+#create SeuratObject
```
```suggestion
```

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+    shell:
+        """
+        if [[ {wildcards.experiment} == *"ANND"* ]]; then
+            Rscript {workflow.basedir}/scripts/convert.R {wildcards.experiment}
```
`convert.R` is not very informative , please rename script

pmb59 commented on scripts/convert.R:
```diff
@@ -0,0 +1,45 @@
+#create SeuratObject
+C = CreateSeuratObject(counts = counts, meta.data = metadata)
+
+#rename colum names. If "inferred_cell_type_._ontology_labels_ontology" is 
```
```suggestion
# rename colum names. If "inferred_cell_type_._ontology_labels_ontology" is 
```

pmb59 commented on scripts/convert.R:
```diff
@@ -0,0 +1,45 @@
+
+#rename colum names. If "inferred_cell_type_._ontology_labels_ontology" is 
+# not present in metadata, try C$inferred_cell_type_._authors_labels_ontology 
+# we want to extract the UBERON ids from the links --> basename
```
what if any is present?

pmb59 commented on scripts/createSeuratObject.R:
```diff
@@ -0,0 +1,38 @@
+suppressMessages(library(stringr))
+
+args <- commandArgs(trailingOnly = TRUE)
+
```
```suggestion
if(length(args) != 1) {
   stop("Not correct number of arguments. Please supply 1 argument")
}
```

pmb59 commented on scripts/createSeuratObject.R:
```diff
@@ -0,0 +1,38 @@
+args <- commandArgs(trailingOnly = TRUE)
+
+sample <- args[1]
+filename_O = paste0('scxa_input/',sample,'/', sample, '.project_seurat.rds')
```
again, please find better names for the variables

pmb59 commented on scripts/genRef_1.R:
```diff
@@ -0,0 +1,65 @@
+## Reference matrix type 1 (rows = genenames, cols = celltypes) generation script
+##
+## @zgr2788
```
?

pmb59 commented on scripts/genRef_1.R:
```diff
@@ -0,0 +1,65 @@
+suppressMessages(library(sparseMatrixStats))
+
+#Read data
+args <- commandArgs(trailingOnly = TRUE)
```
```suggestion
args <- commandArgs(trailingOnly = TRUE)
if(length(args) != 2) {
   stop("Not correct number of arguments. Please supply 2 arguments")
}

```

pmb59 commented on scripts/genRef_1.R:
```diff
@@ -0,0 +1,65 @@
+## Reference matrix type 1 (rows = genenames, cols = celltypes) generation script
```
I think you should add this in all your R scripts
```suggestion
#!/usr/bin/env Rscript

## Reference matrix type 1 (rows = genenames, cols = celltypes) generation script
```

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
+    conda:
+        'envs/refgen.yaml'
+    params:
+        method = "none"
```
this needs some explanation too. Why is it always "none"?

pmb59 commented on scripts/makeSummary.R:
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env Rscript
+### script to create a summary file of all the organism part references we created
```
```suggestion

### script to create a summary file of all the organism part references we created
```

pmb59 commented on scripts/genRef_1.R:
```diff
@@ -0,0 +1,65 @@
+    C_0@assays$RNA@counts <- DESeq2::varianceStabilizingTransformation(C_0@assays$RNA@counts)
+  }
+
+)
```
R scripts can benefit from a style formatting. Example:
```suggestion
switch(transform,
  "none" = {
    message('Warning: No transformation applied')
  },
  "log" = {
    C_0@assays$RNA@counts <- log1p(C_0@assays$RNA@counts)
  },
  "sqrt" = {
    C_0@assays$RNA@counts <- sqrt(C_0@assays$RNA@counts)
  },
  "vst" = {  # do not use
    message('Warning: varianceStabilizingTransformation requires high memory and is not recommended.')
    C_0@assays$RNA@counts <- round(C_0@assays$RNA@counts)
    C_0@assays$RNA@counts <- DESeq2::varianceStabilizingTransformation(C_0@assays$RNA@counts)
  }
)

```

pmb59 commented on scripts/makeSummary.R:
```diff
@@ -0,0 +1,46 @@
+
+# Load required packages
+suppressMessages(library(devtools))
+devtools::install_github("YY-SONG0718/scOntoMatch")
```
```suggestion
if (!require("scOntoMatch")) devtools::install_github("YY-SONG0718/scOntoMatch")
```

pmb59 commented on scripts/makeSummary.R:
```diff
@@ -0,0 +1,46 @@
+suppressMessages(library(stringr))
+
+args <- commandArgs(trailingOnly = TRUE)
+
```
check here number of argument is correct - stop if not

pmb59 commented on scripts/makeSummary.R:
```diff
@@ -0,0 +1,46 @@
+if (!file.exists(outfile)) {
+  write.table(data.frame(Accession = character(), UBERON_ID = character(), 
+                         organism_part = character(), date = character()),
+              outfile, sep = "\t", row.names = FALSE, col.names = TRUE)
```
```suggestion
write.table(data.frame(
  Accession = character(),
  UBERON_ID = character(),
  organism_part = character(),
  date = character()
  ), 
  outfile, sep = "\t", row.names = FALSE, col.names = TRUE)
```
This makes it easier to see the column names and the values being passed to data.frame().

pmb59 commented on scripts/makeSummary.R:
```diff
@@ -0,0 +1,46 @@
+              outfile, sep = "\t", row.names = FALSE, col.names = TRUE)
+}
+
+#read in UBERON ontologys to translate UBERON ids to organism part labels
```
```suggestion
# read in UBERON ontologies to translate UBERON ids to organism part labels
```

pmb59 commented on scripts/makeSummary.R:
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env Rscript
+### script to create a summary file of all the organism part references we created
+
+# Load required packages
```
```suggestion
```
when comments are very obvious, better remove

pmb59 commented on scripts/makeSummary.R:
```diff
@@ -0,0 +1,46 @@
+                              propagate_relationships = propagate_relationships, 
+                              extract_tags = 'everything')
+
+#append the summary file for all the references
```
```suggestion
# append the summary file for all the references
```

pmb59 commented on scripts/makeUMAPplots.R:
```diff
@@ -0,0 +1,36 @@
+seurat_obj <- RunUMAP(seurat_obj, dims = 1:10)
+Idents(seurat_obj) = seurat_obj$cell_type_names
+#plot old cell type labels before running reduceCellTypes.R
+p1 = DimPlot(seurat_obj, 
```
Please use `<-` instead of `=` in the R scripts
```suggestion
p1 <- DimPlot(seurat_obj, 
```

pmb59 commented on scripts/makeUMAPplots.R:
```diff
@@ -0,0 +1,36 @@
+# run basic steps in order to gerneat UMAP plot from SeuratObject
+seurat_obj <- FindVariableFeatures(seurat_obj, selection.method = "vst", nfeatures = 2000)
+seurat_obj <- ScaleData(seurat_obj)
+seurat_obj <- RunPCA(seurat_obj, features = VariableFeatures(object = seurat_obj))
+seurat_obj <- RunUMAP(seurat_obj, dims = 1:10)
```
these steps around can require significan memory if you have many cells.

Maybe let the `get_mem_mb` go to few attempts (reiterations)

pmb59 commented on scripts/reduceCellTypes.R:
```diff
@@ -0,0 +1,111 @@
+
+# Load required packages
+suppressMessages(library(devtools))
+suppressMessages(devtools::install_github("YY-SONG0718/scOntoMatch"))
```
```suggestion

# Load required packages
suppressMessages(library(devtools))
if (!require("scOntoMatch")) devtools::install_github("YY-SONG0718/scOntoMatch")
suppressMessages(library(scOntoMatch))
```

pmb59 commented on scripts/reduceCellTypes.R:
```diff
@@ -0,0 +1,111 @@
+
+# get a list of CL ontologies that we can map CL ids predent in SeuratObject to 
+additional_ontologies = names(ont$ancestors)
+additional_ontologies = unlist(ont$children, use.names = F)
```
```suggestion
additional_ontologies <- unlist(ont$children, use.names = FALSE)
```

pmb59 commented on scripts/splitIntoTissues.R:
```diff
@@ -0,0 +1,54 @@
+  seurat_downsampled[ct] <- seurat_tissue[,cells]
+}
+seurat_downsampled = merge(seurat_downsampled[[1]], 
+                          y = seurat_downsampled[2:length(cell_types)] ) 
```
```suggestion
seurat_downsampled = merge(
  seurat_downsampled[[1]],
  y = seurat_downsampled[2:length(cell_types)]
)
```

pmb59 commented on Snakefile:
```diff
@@ -1 +1,246 @@
 
+#Set config here
+# config should 'accessions' to devoncolve and 'exclude_tissues_from_accessions'
+configfile: 'config.yaml'
```
Maybe you could add a simple example of this file in the repo..

nadjano replied:
Yes, added 'example.config'

pmb59 commented on README.md:
```diff
@@ -1,2 +1,24 @@
 # atlas-deconvolution-references
-Generate single-cell reference dataset for deconvolution in Atlas
+Snakemake workflow to generate single-cell reference dataset for deconvolution in Atlas
```
```suggestion
Snakemake workflow to generate single-cell reference dataset for bulk RNA-seq deconvolution in Atlas
```

pmb59 reviewed (changes requested):
Hi @nadjano  
This is looking great. Mostly I have minor comments to improve formatting and style.

If you have intermediate files that are not needed, I would suggest you mark them as `temp()` so Snakemake can delete when non necessary.

The README could also explain better the inputs and main output of the workflow

nadjano committed:
Update Snakefile

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
reduced number of attempts to 3

nadjano committed:
Remove whitespace

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Removed if/else as it was unnecessary

nadjano committed:
Update scripts/reduceCellTypes.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update scripts/splitIntoTissues.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update README.md

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update Snakefile

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update README.md

nadjano committed:
Update README.md

nadjano committed:
Update envs/UMAPplot.yaml

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update scripts/convert.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update scripts/convert.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update Snakefile

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
make Conda first in each rule

nadjano committed:
Update scripts/convert.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
move rule comment inside rule

nadjano committed:
check if scOntomatch needs to be installed

nadjano committed:
changed = to <-

nadjano committed:
Update convert.R

nadjano committed:
Rename convert.R to createSeuratObjectFromANND.R

nadjano committed:
rename script

nadjano committed:
Update createSeuratObjectFromANND.R

nadjano committed:
Update createSeuratObject.R

nadjano committed:
improved variable names

nadjano committed:
improve var names

nadjano committed:
added missing whitespace

nadjano committed:
Update scripts/genRef_1.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update scripts/makeSummary.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update scripts/makeSummary.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update scripts/genRef_1.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update scripts/genRef_1.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update scripts/makeSummary.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update scripts/makeSummary.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
added logging and set -e to each rule

nadjano committed:
Update scripts/makeSummary.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
added check of args length

nadjano committed:
Update scripts/makeSummary.R

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update and rename env.yaml to createANNDSeuratObject.yaml

nadjano committed:
Delete createANNDSeuratObject.yaml

Not used in snake file

nadjano committed:
added information about ontology files

nadjano committed:
added explanation

nadjano committed:
improved initial comment

nadjano committed:
added Seurat version

nadjano committed:
added seurat version

nadjano committed:
added seurat version

nadjano committed:
added seurat version

nadjano committed:
added Seurat version

nadjano committed:
added species to reference dir

nadjano committed:
Update Snakefile

nadjano committed:
added species

nadjano committed:
typo

nadjano committed:
wrong column name

nadjano committed:
Update createSeuratObjectFromANND.R

nadjano committed:
add r-stringi

nadjano committed:
added   r-stringi

nadjano committed:
added  r-stringi

nadjano committed:
added  - r-stringi

nadjano committed:
added  - r-stringi

nadjano committed:
clarify how to proceede which duplicated reference

nadjano committed:
change atlas_prod to sc_exps

nadjano committed:
correct path for file

nadjano committed:
correct path of obo file

nadjano committed:
correct path of obo file

nadjano committed:
store CL terms instead of IDs in CellType column

pmb59 commented on README.md:
```diff
@@ -1,2 +1,62 @@
+* one png file per scxa accession + organism part with two UMAP plots (old + new cell type labels) for quality control
+
+## Contributions
+Folowing scripts are taken from the [CATD_snakemake pipeline](https://github.com/Functional-Genomics/CATD_snakemake) which benchmarks deconvolution methods:
```
```suggestion
The following scripts are taken from the [CATD_snakemake pipeline](https://github.com/Functional-Genomics/CATD_snakemake) which benchmarks deconvolution methods:
```

pmb59 commented on Snakefile:
```diff
@@ -1 +1,272 @@
+## Reference library preperation for deconvolution in 
+## Expression Atlas
+##
+## 
+##
+##
```
```suggestion
```

nadjano committed:
Update Snakefile

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
Update README.md

Co-authored-by: Pedro Madrigal <8195212+pmb59@users.noreply.github.com>

nadjano committed:
remove unnesessary

nadjano committed:
Update README.md

nadjano committed:
add species 

enable usage of one config file for different species

nadjano committed:
add species wildcard to createSeuratObject

nadjano committed:
add species information

nadjano committed:
add species information

nadjano committed:
add missing '/'

nadjano committed:
remove duplicated line

nadjano committed:
add check of args length

nadjano committed:
Update example.config for multiple species

nadjano committed:
remove downsampling as it led to errors

nadjano committed:
add downsampling part again

nadjano committed:
change order to avoid problems for tissues without cells

nadjano committed:
only select tissues which have cell type ontology labels

error where caused by missing cell type ontologies

nadjano committed:
& instead of &&

nadjano committed:
missing #

nadjano committed:
account for organism parts with only one cell type

nadjano committed:
avoid organism parts with less than two cell types or not enough cells per cell type

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
correct path names

nadjano committed:
correct path names

nadjano committed:
Update Snakefile

nadjano committed:
args length of 2 is enough

nadjano committed:
take care of other column names

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
use references instead of UMAPs as last input

nadjano committed:
fix to_remove tissues

nadjano committed:
add UMAP output to last rule

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
change max number of cell types to avoid cell type reduction

nadjano committed:
Update reduceCellTypes.R

nadjano committed:
Update createSeuratObject.R

nadjano committed:
Update createSeuratObjectFromANND.R

nadjano committed:
Update createSeuratObjectFromANND.R

nadjano committed:
add missing '0'

nadjano committed:
Update Snakefile

nadjano committed:
Update createSeuratObject.R

nadjano committed:
Update createSeuratObjectFromANND.R

nadjano committed:
Update Snakefile

nadjano committed:
get files from atlas-prod instead of copying

nadjano committed:
switch rows and cols files

nadjano committed:
use direct paths instead of copied files

nadjano committed:
Update createSeuratObject.R

nadjano committed:
allow more arguments

nadjano committed:
allow more arguments

nadjano committed:
Update createSeuratObjectFromANND.R

nadjano committed:
Update reduceCellTypes.R

nadjano committed:
change to using condensed sdrf instead of cell_metadata

nadjano committed:
Update Snakefile

nadjano committed:
get cell metadata from condensed sdrf

cell-metadata file has not all zoom mappings sometimes therefore change to condensed sdrf file.

nadjano committed:
read condensed-sdrf.tsv instead of  cell-metadata for non ANND

nadjano committed:
check if uberon is from organism part

nadjano committed:
Update Snakefile

nadjano committed:
Update Snakefile

nadjano committed:
Update createSeuratObject.R

nadjano committed:
Update createSeuratObject.R

nadjano committed:
Update reduceCellTypes.R

pmb59 committed:
rm few empty lines

pmb59 reviewed (approved):
LGTM @nadjano !

thanks for addressing my comments
